src/flownet/ahm/_run_ahm.py

```diff
@@ -416,14 +416,23 @@ def run_flownet_history_matching(
         field_data.faults if config.model_parameters.fault_mult else None
     )
 
-    concave_hull_bounding_boxes: Optional[np.ndarray] = None
+    concave_hull_list: Optional[List[np.ndarray]] = None
     if config.flownet.data_source.concave_hull:
-        concave_hull_bounding_boxes = field_data.grid_cell_bounding_boxes
+        concave_hull_list = []
+        if df_well_connections["LAYER_ID"].nunique() > 1:
```

Comment:
Is the if-statement actually needed? (Are LayerID's also assigned when there is only 1 layer; if so you could drop the if statement and always go through the loop)

Reply:
I think you are right. We always assign a layerID, if no layering is defined they just all have the same layerID

Reply:
Ok, then its not needed (anymore).